### ISSUE TITLE
Please support Typescript

### ISSUE THREAD

**Issue by trashhalo** [codeclimate/codeclimate-duplication#127]
## [Typescript](https://www.typescriptlang.org/)

* [Typescript ships lots of tools to support writing linters and formatters](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#traversing-the-ast-with-a-little-linter)

* [Its backed by Microsoft](https://blogs.msdn.microsoft.com/typescript/) and [Google](https://techcrunch.com/2015/03/05/microsoft-and-google-collaborate-on-typescript-hell-has-not-frozen-over-yet/)

* [Its more searched for than coffeescript](https://www.google.com/trends/explore#q=typescript%2C%20coffeescript&date=1%2F2012%2049m&cmpt=q&tz=Etc%2FGMT%2B4)

**Comment by XMichaelMichaelX** [comment]
would really love to have this feature implemented in CodeClimate

**Comment by dblandin** [comment]
Hey @trashhalo @XMichaelMichaelX,

We now have a beta Typescript engine available, maintained by @tkqubo!

https://github.com/tkqubo/codeclimate-tslint

You can use it on https://codeclimate.com or via the CLI with the following addition in your `.codeclimate.yml` configuration file:

```yaml
engines:
  tslint:
    enabled: true
    channel: beta
```

**Comment by rraziel** [comment]
Just some feedback, but the duplication analysis seems... somewhat dubious:

![image](https://user-images.githubusercontent.com/3615018/33253803-aa1f4cba-d345-11e7-8444-d86618ed889d.png)

Anything I may have misconfigured?

**Comment by zenspider** [comment]
This is supported. Please close. Open new tickets for separate issues so they don't get buried.

**Comment by another-guy** [comment]
@zenspider I'm having hard time finding documentation on TypeScript support. May I kindly ask you to point to the docs before closing the issue?

**Comment by zenspider** [comment]
Sorry. I'm just a contractor for codeclimate. I don't actually know where any of the doco is. I just fix stuff. 😛 

That said, I did just fix some of the stuff around both javascript and typescript. If you add it to your enabled `languages`, it should Just Work™. Example:

```yaml
engines:
  duplication:
    enabled: true
    config:
      languages:
      - javascript
      - typescript
```

**Comment by Lorel** [comment]
Indeed, _duplication_ does support _Typescript_: https://github.com/codeclimate/codeclimate-duplication/blob/eec0293fdb66436154272af2f750f2b12e26dcc5/lib/cc/engine/duplication.rb#L34
So this issue should be closed

But actually, I didn't find out any mention of all the supported languages here
_Duplication_ is a very great tool, we should mention this clearly to encourage people of using it, this one worth it
